// vue-socials/dist/esm/components/SWeibo/SWeibo.js
import Vue from 'vue';
import BaseSocials from '../../mixins/BaseSocial/BaseSocial.js';
import getSerialisedParams from '../../utils/getSerialisedParams.js';

/**
 * Hey!
 *
 * SWeibo component used for Weibo social network
 * @link https://weibo.com/
 */
/**
 * Share parameters for link
 * @link http://open.weibo.com/wiki/ShareCode
 */

var SWeibo = /* #__PURE__ */Vue.extend({
  name: 'SWeibo',
  mixins: [BaseSocials('Weibo')],
  computed: {
    networkURL: function networkURL() {
      var BASE_URL = 'http://service.weibo.com/share/share.php';
      var shareOptions = this.shareOptions;
      var url = shareOptions.url,
          title = shareOptions.title,
          appkey = shareOptions.appkey,
          pic = shareOptions.pic;
      var serialisedParams = getSerialisedParams({
        url: url,
        title: title,
        appkey: appkey,
        pic: pic
      });
      return "".concat(BASE_URL).concat(serialisedParams);
    }
  },
  render: function render(h) {
    return this.generateComponent(h, this.networkURL);
  }
});

export default SWeibo;
